Documentation/ceph-cluster-crd.md

```diff
@@ -1319,128 +1319,63 @@ The features available from the external cluster will vary depending on the vers
-
-Finally, you can simply execute the script like this from a machine that has access to your Kubernetes cluster:
-
+rbd-data-pool-name should be created, and if you need to have CephFS secret details you have to pass filesystem name, and if you need to create rgw_admin_ops_user you need to pass rgw_endpoint (Rados GateWay endpoint (in <IP>:<PORT> format)) , and if you need to export rgw-pool-prefix prepare rgw pools
+example: 
 ```console
-bash deploy/examples/import-external-cluster.sh
+python3 create-external-cluster-resources.py --rbd-data-pool-name <pool_name> --cephfs-filesystem-name <filesystem-name> --rgw-endpoint  <rgw-endpoint> --namespace <namespace> --rgw-pool-prefix <rgw-pool-prefix> --format bash
 ```
 
-#### CephCluster example (consumer)
+>Note: Use format bash, as to keep up with the further steps, and rbd-data-pool and cephfs-filesystem, should be created in advance.
 
-Assuming the above section has successfully completed, here is a CR example:
+2. After that you need to copy the bash output.
 
-```yaml
-apiVersion: ceph.rook.io/v1
-kind: CephCluster
-metadata:
-  name: rook-ceph-external
-  namespace: rook-ceph-external
-spec:
-  external:
-    enable: true
-  crashCollector:
-    disable: true
-  # optionally, the ceph-mgr IP address can be pass to gather metric from the prometheus exporter
-  #monitoring:
-    #enabled: true
-    #rulesNamespace: rook-ceph
-    #externalMgrEndpoints:
-      #- ip: 192.168.39.182
-    #externalMgrPrometheusPort: 9283
-```
+#### Run commands on machine that has access to your Kubernetes cluster:
 
-Choose the namespace carefully, if you have an existing cluster managed by Rook, you have likely already injected `common.yaml`.
-Additionally, you now need to inject `common-external.yaml` too.
+1. Deploy Rook-Ceph, create [common.yaml](/deploy/examples/common.yaml), [crds.yaml](/deploy/examples/crds.yaml) and [operator.yaml](/deploy/examples/operator.yaml) manifests.
 
-You can now create it like this:
+2. Just paste the above output from `create-external-cluster-resources.py` into your current shell, it will export all the data.
 
+3. Run the [import](/deploy/examples/import-external-cluster.sh) script.
 ```console
-kubectl create -f deploy/examples/cluster-external.yaml
+. import-external-cluster.sh
 ```
 
-If the previous section has not been completed, the Rook Operator will still acknowledge the CR creation but will wait forever to receive connection information.
-
-> **WARNING**: If no cluster is managed by the current Rook Operator, you need to inject `common.yaml`, then modify `cluster-external.yaml` and specify `rook-ceph` as `namespace`.
-
-If this is successful you will see the CepCluster status as connected.
+4. Create [common-external.yaml](/deploy/examples/common-external.yaml) and [cluster-external.yaml](/deploy/examples/cluster-external.yaml)
```

Comment:
Oh, do we need both common.yaml and common-external.yaml? I thought we only needed one of them.

Reply:
Right, we would need both if you have both external and local clusters, but the common scenario that we should document is when there is only an external cluster config. Do we not need common.yaml in that case?

Reply:
I need to test, 
But according to https://github.com/rook/rook/blob/55a8fb9b8e1e89767ee70a0e16ae2fbc12c71ba2/deploy/examples/common-external.yaml#L2, 
This I can say is `common.yaml`  is needed and `common-external.yaml` is only needed in case of multi-tenant, but if we not use `rook-ceph as the namespace`

Reply:
In a follow-up issue I'd like to clean this up so we only need to start the operator in the rook-ceph-external namespace and not create all the RBAC in common.yaml

Reply:
okay